pkg/konnector/controllers/cluster/servicebinding/servicebinding_reconcile.go

```diff
@@ -46,7 +50,38 @@ type reconciler struct {
 }
 
 func (r *reconciler) reconcile(ctx context.Context, binding *kubebindv1alpha1.APIServiceBinding) error {
+	logger := klog.FromContext(ctx)
+
 	var errs []error
+	var kubeconfig string
+
+	ref := binding.Spec.KubeconfigSecretRef
+	secret, err := r.getSecret(ref.Namespace, ref.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	} else if errors.IsNotFound(err) {
+		logger.V(2).Info("secret not found", "secret", ref.Namespace+"/"+ref.Name)
+	} else {
+		kubeconfig = string(secret.Data[ref.Key])
+	}
+
+	// extract which namespace this kubeconfig points to
+	cfg, err := clientcmd.Load([]byte(kubeconfig))
+	if err != nil {
+		logger.Error(err, "invalid kubeconfig in secret", "namespace", ref.Namespace, "name", ref.Name)
+		return nil // nothing we can do here. The APIServiceBinding Controller will set a condition
+	}
+	kubeContext, found := cfg.Contexts[cfg.CurrentContext]
+	if !found {
+		logger.Error(err, "kubeconfig in secret does not have a current context", "namespace", ref.Namespace, "name", ref.Name)
+		return nil // nothing we can do here. The APIServiceBinding Controller will set a condition
+	}
+
+	// As konnector is running APIServiceBinding controller for each provider cluster,
+	// so each controller should skip others provider's APIServiceBinding object
+	if kubeContext.Namespace != r.providerNamespace {
+		return nil
+	}
```

Comment:
wondering, is this even enough? Couldn't there be multiple connected provider clusters with the same namespace name? I would rather store the APIServiceBinding name here in the controller instance and skip other service bindings.

Reply:
yes, there can be multiple clusters with the same namespace name. so it is better to keep the service binding name in controller instance. Updating...
Thanks!

Reply:
Updated with service binding name.